src/ResourceManagement/OperationalInsights/OperationalInsights.Test/ScenarioTests/DataSourceOperationsTests.Scenario.cs

```diff
@@ -0,0 +1,178 @@
+                {
+                    Location = resourceGroup.Location,
+                    Sku = new Sku(SkuNameEnum.Standard)
+                };
+
+                var workspaceResponse = client.Workspaces.CreateOrUpdate(
+                    resourceGroupName,
+                    workspaceName,
+                    workspace);
+
+                TestHelper.ValidateWorkspace(workspace, workspaceResponse);
+
+                // Create 200+ dataSource
+                for (int i = 0; i < 300; i++)
+                {
+                    string windowsEventDataSourceName = TestUtilities.GenerateName("AzTestDSWE");
+                    var createParameters = new DataSource
+                    {
+                        Kind = DataSourceKind.WindowsEvent,
+                        Properties = JToken.Parse("{\"eventLogName\": \"" + ("windowsEvent" + i) + "\", \"eventTypes\": [{\"eventType\": \"Error\"}]}")
+                    };
+
+                    var createResponse = client.DataSources.CreateOrUpdate(resourceGroupName, workspaceName, windowsEventDataSourceName, createParameters);
+                    TestHelper.ValidateDatasource(createParameters, createResponse);
+                }
+
+                // List the data sources in the workspace
+                var listResponse = client.DataSources.ListByWorkspace(new ODataQuery<DataSourceFilter>(ds => ds.Kind == "WindowsEvent"), resourceGroupName, workspaceName);
+                Assert.Equal(200, listResponse.Count());
+                Assert.NotNull(listResponse.NextPageLink);
+
+                // Get the next page
+                var listResponseNextPage = client.DataSources.ListByWorkspaceNext(listResponse.NextPageLink);
+                Assert.Null(listResponseNextPage.NextPageLink);
+                Assert.Equal(100, listResponseNextPage.Count());
+            }
+        }
+
+        [Fact]
+        public void DataSourceCreateFailsWithoutWorkspace()
+        {
+            using (MockContext context = MockContext.Start(GetType().FullName))
+            {
+                var resourceClient = TestHelper.GetResourceManagementClient(this, context);
+                var client = TestHelper.GetOperationalInsightsManagementClient(this, context);
+
+                string resourceGroupName = TestUtilities.GenerateName("OIAutoRest");
+                var resourceGroup = TestHelper.CreateResourceGroup(resourceGroupName, resourceClient);
+
+                // Create a data source on a non-existent workspace
+                string dataSourceName = TestUtilities.GenerateName("AzTestDS");
+                string workspaceName = TestUtilities.GenerateName("AzTest");
+
+                var createParameters = new DataSource
+                {
+                    Kind = "AzureAuditLog",
+                    Properties = JToken.Parse("{'LinkedResourceId':'/subscriptions/a6383be3-f0e8-4968-93d5-10f2625f5bb5/providers/microsoft.insights/eventtypes/management'}")
```

Comment:
This seems awkward.  Why is this not represented as a property of the CreateParameters?  Also, why is this buried under CreateParameters.Properties, ratehr than being a top level property?